iroh-net/src/magicsock.rs

```diff
@@ -348,18 +350,42 @@ impl Inner {
                             udp_sent = true;
                             // record metrics.
                         }
-                        Err(err) => {
+                        Poll::Ready(Err(err)) => {
                             error!(node = %public_key.fmt_short(), ?addr, "failed to send udp: {err:?}");
                             udp_error = Some(err);
                         }
+                        Poll::Pending => {
+                            udp_pending = true;
+                        }
                     }
                 }
 
                 // send derp
                 if let Some(ref derp_url) = derp_url {
-                    self.try_send_derp(derp_url, public_key, split_packets(&transmits));
-                    transmits_sent = transmits.len();
-                    derp_sent = true;
+                    match self.try_send_derp(derp_url, public_key, split_packets(&transmits)) {
+                        Poll::Ready(sent) => {
+                            derp_sent = sent;
+                            transmits_sent = transmits.len();
+                        }
+                        Poll::Pending => {
+                            self.network_send_wakers.lock().replace(cx.waker().clone());
```

Comment:
huh, the derp actor already wakes this.  I guess this was always intended to be like this but was somehow overlooked?

Reply:
it got lost in a refactoring I think